vnext/Shared/Networking/RedirectHttpFilter.cpp

```diff
@@ -0,0 +1,281 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#undef WINRT_LEAN_AND_MEAN
+
+#include "RedirectHttpFilter.h"
+
+#include "WinRTTypes.h"
+
+// Windows API
+#include <winapifamily.h>
+#include <winrt/Windows.Web.Http.Headers.h>
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+#include <WinInet.h>
+#else
+#define INTERNET_ERROR_BASE 12000
+#define ERROR_HTTP_REDIRECT_FAILED (INTERNET_ERROR_BASE + 156)
+#endif
+
+using winrt::Windows::Foundation::Uri;
+using winrt::Windows::Foundation::Collections::IVector;
+using winrt::Windows::Security::Credentials::PasswordCredential;
+using winrt::Windows::Security::Cryptography::Certificates::Certificate;
+using winrt::Windows::Security::Cryptography::Certificates::ChainValidationResult;
+using winrt::Windows::Web::Http::HttpMethod;
+using winrt::Windows::Web::Http::HttpRequestMessage;
+using winrt::Windows::Web::Http::HttpResponseMessage;
+using winrt::Windows::Web::Http::HttpStatusCode;
+using winrt::Windows::Web::Http::Filters::IHttpBaseProtocolFilter;
+using winrt::Windows::Web::Http::Filters::IHttpFilter;
+
+namespace Microsoft::React::Networking {
+
+#pragma region RedirectHttpFilter
+
+RedirectHttpFilter::RedirectHttpFilter(
+    size_t maxRedirects,
+    IHttpFilter &&innerFilter,
+    IHttpFilter &&innerFilterWithNoCredentials) noexcept
+    : m_maximumRedirects{maxRedirects},
+      m_innerFilter{std::move(innerFilter)},
+      m_innerFilterWithNoCredentials{std::move(innerFilterWithNoCredentials)} {
+  // Prevent automatic redirections.
+  if (auto baseFilter = m_innerFilter.try_as<IHttpBaseProtocolFilter>()) {
+    baseFilter.AllowAutoRedirect(false);
+    baseFilter.AllowUI(false);
+  }
+  if (auto baseFilter = m_innerFilterWithNoCredentials.try_as<IHttpBaseProtocolFilter>()) {
+    baseFilter.AllowAutoRedirect(false);
+    baseFilter.AllowUI(false);
+  }
+}
+
+RedirectHttpFilter::RedirectHttpFilter(IHttpFilter &&innerFilter, IHttpFilter &&innerFilterWithNoCredentials) noexcept
+    : RedirectHttpFilter(20, std::move(innerFilter), std::move(innerFilterWithNoCredentials)) {}
```

Comment:
nit, make a constexpr size_t DefaultMaxRedirects = 20 and use it here so it's clearer as to what 20 means